File: Miscellaneous/Golang/387. First Unique Character in a String.go
package main

import (
	"fmt"
	"strings"
)

func main() {

	str := "loveleetcode"
	result := uniqueChar(str)
	fmt.Println(result)
}

func uniqueChar(s string) int {
	str := strings.Split(s, "")
	ansMap := make(map[string]int)

	for _, char := range str {
		ansMap[char]++
	}

	for i := 0; i < len(str); i++ {
		val, ok := ansMap[str[i]]
		if ok && val == 1 {
			return i
		}
	}

	return -1
}

File: Miscellaneous/Python/387. First Unique Character in a String.py
class Solution:
    def uniqueChar(self, str):
        dict = {}

        for char in str:
            if char not in dict:
                dict[char] = 1
            else:
                dict[char] += 1
        
        
        for i in range(len(str)):
            if dict[str[i]] == 1:
                return i
        
        return -1

obj = Solution()
str = "loveleetcode"
ans = obj.uniqueChar(str)
print(ans)